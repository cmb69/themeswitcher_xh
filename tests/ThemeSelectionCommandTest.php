<?php

/**
 * Copyright (C) 2023 Christoph M. Becker
 *
 * This file is part of Themeswitcher_XH.
 *
 * Themeswitcher_XH is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Themeswitcher_XH is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Themeswitcher_XH.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace Themeswitcher;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Themeswitcher\Infra\View;

class ThemeSelectionCommandTest extends TestCase
{
    public function testRendersThemeSelector(): void
    {
        global $cf;
        $cf = ["site" => ["template" => "foo_theme"]];
        $model = $this->createMock(Model::class);
        $model->method("getThemes")->willReturn(["foo_theme", "bar_template"]);
        $view = new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["themeswitcher"]);
        $sut = new ThemeSelectionCommand($model, $view);
        $response = $sut->render();
        Approvals::verifyHtml($response);
    }
}
