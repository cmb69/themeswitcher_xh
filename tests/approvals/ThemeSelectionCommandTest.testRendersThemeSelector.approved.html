<form class="themeswitcher_select_form" method="get">
    <input type="hidden" name="selected" value="">
    <label>
        <span>Theme</span>
        <select name="themeswitcher_select" onchange="this.form.submit()">
            <option value="foo_theme" selected>foo_theme</option>
            <option value="bar_template" >bar_template</option>
        </select>
    </label>
    <button class="themeswitcher_button">Activate</button>
</form>
<script>
    const each = Array.prototype.forEach;
    each.call(document.getElementsByClassName("themeswitcher_button"), function (button) {
        button.style.display = "none";
    });
</script>
